File: JITSDataTools/src/models/Point.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package models;

import utilities.Parameters;

/**
 *
 * @author Luzec
 */
public class Point {
    private Double lat;
    private Double lng;

    public Point(Double lat, Double lng) {
        this.lat = lat;
        this.lng = lng;
    }

    public Point(String lat, String lng)
    {
        this.lat = Double.valueOf(lat);
        this.lng = Double.valueOf(lng);
    }
    
    public Double getLat() {
        return lat;
    }

    public void setLat(Double lat) {
        this.lat = lat;
    }

    public Double getLng() {
        return lng;
    }

    public void setLng(Double lng) {
        this.lng = lng;
    }
    
    // Distance in km unit
    public double distance(Point point) {
        double lat1 = this.lat;
        double lng1 = this.lng;
        double lat2 = point.lat;
        double lng2 = point.lng;
        double earthRadius = Parameters.R;
        double dLat = Math.toRadians(lat2 - lat1);
        double dLng = Math.toRadians(lng2 - lng1);
        double sindLat = Math.sin(dLat / 2);
        double sindLng = Math.sin(dLng / 2);
        double a = Math.pow(sindLat, 2)
                        + Math.pow(sindLng, 2) * Math.cos(Math.toRadians(lat1)) * Math.cos(Math.toRadians(lat2));
        double c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
        double dist = earthRadius * c;
        return dist;
    }
}

File: DataTools/DataTools/Models/Point.cs
using DataTools.Utilities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataTools.Models
{
    class Point
    {
        public Double lat { set; get; }
        public Double lng { set; get; }

        public Point(double lat, double lng)
        {
            this.lat = lat;
            this.lng = lng;
        }

        public Point(string lat, string lng)
        {
            this.lat = Double.Parse(lat);
            this.lng = Double.Parse(lng);
        }

        public double Distance(Point point)
        {
            double lat1 = this.lat; double lng1 = this.lng; double lat2 = point.lat; double lng2 = point.lng;

            double earthRadius = Parameters.R; //6371 * 1000
            double dLat = MathLib.ToRadians(lat2 - lat1);
            double dLng = MathLib.ToRadians(lng2 - lng1);
            double sindLat = Math.Sin(dLat / 2);
            double sindLng = Math.Sin(dLng / 2);
            double a = Math.Pow(sindLat, 2)
                    + Math.Pow(sindLng, 2) * Math.Cos(MathLib.ToRadians(lat1)) * Math.Cos(MathLib.ToRadians(lat2));
            double c = 2 * Math.Atan2(Math.Sqrt(a), Math.Sqrt(1 - a));
            double dist = earthRadius * c;
            return dist;
        }
    }
}
